internal/app/init/main.go

```diff
@@ -175,25 +178,39 @@ func root() (err error) {
 		}
 	}
 
-	go func() {
-		if err := listenForPowerButton(); err != nil {
-			log.Printf("WARNING: power off events will be ignored: %+v", err)
-		}
-	}()
+	poweroffCh, err := listenForPowerButton()
+	if err != nil {
+		log.Printf("WARNING: power off events will be ignored: %+v", err)
+	}
+
+	startupErrCh := make(chan error)
+	termCh := make(chan os.Signal, 1)
+	signal.Notify(termCh, syscall.SIGTERM)
```

Comment:
this works nice with talos-in-docker, but might be also used to initiate reboot at any time by `kill(1, TERM)`